### PR TITLE
Migrate Avatar, img, header, truncatetext

### PR DESCRIPTION
**Pull request by HerrTopi**

### CODE REVIEW

File: packages/ui-img/src/Img/index.js [hunk 1]
```diff
@@ -115,14 +111,11 @@ class Img extends Component {
     const blur = `blur(${this.theme.imageBlurAmount})`
```
Comment (ke1k0) [inline packages/ui-img/src/Img/index.js:111]: This won't work, probably should get it from the style object.

File: packages/ui-img/src/Img/theme.js [hunk 2]
```diff
@@ -22,9 +22,21 @@
...
+ * @return {Object} The final theme object with the overrides and component variables
+ */
+const generateComponentTheme = (theme, themeOverride = {}) => {
+  const componentVariables = {
+    effectTransitionDuration: '1s'
+  }
+
   return {
-    effectTransitionDuration: '1s',
-    imageBlurAmount: '0.25em'
```
Comment (ke1k0) [inline packages/ui-img/src/Img/theme.js:28]: imageBlurAmount missing

File: packages/ui-truncate-text/src/TruncateText/index.js [hunk 3]
```diff
@@ -105,6 +104,8 @@ class TruncateText extends Component {
   constructor(props) {
     super(props)
     this.state = this.initialState
+    // eslint-disable-next-line react/prop-types
```
Comment (ke1k0) [inline packages/ui-truncate-text/src/TruncateText/index.js:107]: Please add it to propTypes and defaultProps.

File: packages/ui-truncate-text/src/TruncateText/theme.js [hunk 4]
```diff
@@ -22,9 +22,24 @@
...
+ * @param  {Object} theme The actual theme object.
+ * @param  {Object} themeOverride User provided overrides of the default theme mapping.
+ * @return {Object} The final theme object with the overrides and component variables
+ */
+const generateComponentTheme = (theme, themeOverride = {}) => {
+  const { typography } = theme
+
+  const componentVariables = {
     fontFamily: typography.fontFamily,
     lineHeight: typography.lineHeight
```
Comment (ke1k0) [inline packages/ui-truncate-text/src/TruncateText/theme.js:36]:
```suggestion
    fontFamily: typography?.fontFamily,
    lineHeight: typography?.lineHeight
```

File: packages/ui-avatar/src/Avatar/index.js [hunk 5]
```diff
@@ -30,6 +30,7 @@ import { View } from '@instructure/ui-view'
 import { ThemeablePropTypes } from '@instructure/ui-themeable'
 import { passthroughProps } from '@instructure/ui-react-utils'
 import { testable } from '@instructure/ui-testable'
+// eslint-disable-next-line import/named
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-avatar/src/Avatar/index.js:33]: Eslint drops error with: "withStyle not found in '@instructure/emotion'"

[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-avatar/src/Avatar/index.js:33]: ESLint warns about: "withStyle not found in '@instructure/emotion'"

File: packages/ui-heading/src/Heading/index.js [hunk 6]
```diff
@@ -21,35 +21,34 @@
...
-import { themeable, ThemeablePropTypes } from '@instructure/ui-themeable'
+import { ThemeablePropTypes } from '@instructure/ui-themeable'
 import { childrenOrValue } from '@instructure/ui-prop-types'
 import {
   deprecated,
   getElementType,
   passthroughProps
 } from '@instructure/ui-react-utils'
 import { testable } from '@instructure/ui-testable'
+// eslint-disable-next-line import/named
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-heading/src/Heading/index.js:37]: ESLint warns about: "withStyle not found in '@instructure/emotion'"

File: packages/ui-avatar/src/Avatar/index.js [hunk 7]
```diff
@@ -22,6 +21,7 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
+/* eslint-disable react/prop-types */
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-avatar/src/Avatar/index.js:24]: Good catch, thanks

File: packages/ui-heading/src/Heading/index.js [hunk 8]
```diff
@@ -116,9 +115,13 @@ class Heading extends Component {
       margin,
       elementRef,
       ellipsis,
+      // eslint-disable-next-line react/prop-types
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-heading/src/Heading/index.js:118]: Thanks, fixed

File: packages/ui-heading/src/Heading/index.js [hunk 9]
```diff
@@ -116,9 +115,13 @@ class Heading extends Component {
...
+      makeStyles,
       ...props
     } = this.props
 
+    const styles = makeStyles(this.state)
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-heading/src/Heading/index.js:123]: Moved :)

File: packages/ui-img/src/Img/index.js [hunk 10]
```diff
@@ -21,33 +21,42 @@
...
+    super(props)
+    this.styles = props.makeStyles(this.state)
+  }
+
+  componentDidUpdate() {
+    this.styles = this.props.makeStyles(this.state)
+  }
+
   static propTypes = {
+    makeStyles: PropTypes.any,
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-img/src/Img/index.js:59]: True. Noted, fixed

File: packages/ui-img/src/Img/index.js [hunk 11]
```diff
@@ -21,33 +21,42 @@
...
 import PropTypes from 'prop-types'
-import classnames from 'classnames'
 
 import { View } from '@instructure/ui-view'
-import { themeable, ThemeablePropTypes } from '@instructure/ui-themeable'
+import { ThemeablePropTypes } from '@instructure/ui-themeable'
 import { passthroughProps, deprecated } from '@instructure/ui-react-utils'
 import { supportsObjectFit } from '@instructure/ui-dom-utils'
-import { testable } from '@instructure/ui-testable'
+// eslint-disable-next-line import/named
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-img/src/Img/index.js:32]: ESLint drops error

File: packages/ui-img/src/Img/index.js [hunk 12]
```diff
@@ -153,19 +143,7 @@ class Img extends Component {
...
 
     const imageProps = {
-      className: classnames({
-        [styles.image]: true,
-        [styles['has-overlay']]: overlay,
-        [styles['has-filter']]: withBlur || withGrayscale || blur || grayscale,
-        [styles.cover]: this.supportsObjectFit && constrain === 'cover',
-        [styles.contain]: this.supportsObjectFit && constrain === 'contain'
-      }),
-      style: {
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-img/src/Img/index.js:163]: It should. Am I missing something? This block applies to an  tag, so it's not related to <View>

The only style related to <View> is backgroundImage, which has no conflict with any of the <View> props.

File: packages/ui-heading/src/Heading/styles.js [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+    bottom: {
+      paddingBottom: componentTheme.borderPadding,
+      borderBottom: `${componentTheme.borderWidth} ${componentTheme.borderStyle} ${componentTheme.borderColor}`
+    }
+  }
+  return {
+    root: {
+      label: 'root',
+      lineHeight: componentTheme.lineHeight,
+      margin: 0,
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-heading/src/Heading/styles.js:105]: Yeah, checked, fixed

File: packages/ui-heading/src/Heading/styles.js [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+      borderTop: `${componentTheme.borderWidth} ${componentTheme.borderStyle} ${componentTheme.borderColor}`
+    },
+    bottom: {
+      paddingBottom: componentTheme.borderPadding,
+      borderBottom: `${componentTheme.borderWidth} ${componentTheme.borderStyle} ${componentTheme.borderColor}`
+    }
+  }
+  return {
+    root: {
+      label: 'root',
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-heading/src/Heading/styles.js:103]: Did it, also I renamed root to heading

File: packages/ui-heading/src/Heading/styles.js [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+
+  const borderStyles = {
+    top: {
+      paddingTop: componentTheme.borderPadding,
+      borderTop: `${componentTheme.borderWidth} ${componentTheme.borderStyle} ${componentTheme.borderColor}`
+    },
+    bottom: {
+      paddingBottom: componentTheme.borderPadding,
+      borderBottom: `${componentTheme.borderWidth} ${componentTheme.borderStyle} ${componentTheme.borderColor}`
+    }
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-heading/src/Heading/styles.js:99]: Weird either way, but I lean towards your suggestion

File: packages/ui-heading/src/Heading/styles.js [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+      lineHeight: componentTheme.lineHeight,
+      margin: 0,
+      ...levelStyles[level],
+      ...colorStyles[color],
+      ...ellipsisStyle,
+      ...borderStyles[border]
+    }
+  }
+}
+export default generateStyle
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-heading/src/Heading/styles.js:113]: Yeah, missed that. Hope it'll work the way I fixed it

File: packages/ui-img/src/Img/theme.js (hunk 2 above)
Comment (HerrTopi) [reply packages/ui-img/src/Img/theme.js:28]: I thought I can get way with it. I didn't think it through properly

File: packages/ui-truncate-text/src/TruncateText/index.js (hunk 3 above)
Comment (HerrTopi) [reply packages/ui-truncate-text/src/TruncateText/index.js:107]: Added

File: packages/ui-truncate-text/src/TruncateText/theme.js (hunk 4 above)
Comment (HerrTopi) [reply packages/ui-truncate-text/src/TruncateText/theme.js:36]: Noted

**Pull request closed without merging**